readers/api/transport.go

```diff
@@ -207,25 +209,34 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 	}
 }
 
-func authorize(r *http.Request, chanID string) error {
-	token := r.Header.Get("Authorization")
-	if token == "" {
-		return errors.ErrAuthentication
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-
-	_, err := auth.CanAccessByKey(ctx, &mainflux.AccessByKeyReq{Token: token, ChanID: chanID})
-	if err != nil {
-		e, ok := status.FromError(err)
-		if ok && e.Code() == codes.PermissionDenied {
-			return errors.ErrAuthorization
+func authorize(ctx context.Context, req listMessagesReq, tc mainflux.ThingsServiceClient, ac mainflux.AuthServiceClient) (err error) {
+	switch {
+	case strings.HasPrefix(req.token, userToken):
+		token := strings.TrimPrefix(req.token, userToken)
+		user, err := usersAuth.Identify(ctx, &mainflux.Token{Value: token})
+		if err != nil {
+			e, ok := status.FromError(err)
```

Comment:
Inline with the line below

Reply:
Actually, since `user` is used later, there is no need to inline. This is just fine:
```go

user, err = usersAuth.Identify(ctx, &mainflux.Token{Value: token})
if err != nil {
...}
use user here
....
```